Build error on old Macs

**Describe the bug**

When building from source with BSD sed, following error happens:

```
./src/version.h:1:1: error: unknown type name 'x23define'
x23define JQ_VERSION "1.8.0"
```

I think the bug was brought by 
https://github.com/jqlang/jq/blob/62cafa2367d87dc70c215e3e3d22118d1a6ddf3d/Makefile.am#L102

If we change `\x23` to `\#` everything works fine.

**To Reproduce**

Build with BSD sed, like `/usr/bin/sed` on macOS.

**Expected behavior**

Build successfully.

**Environment (please complete the following information):**

- OS and Version: macOS
- jq version 1.8

**Additional context**

--- a/Makefile.am
+++ b/Makefile.am
@@ -99,7 +99,7 @@ endif
 # Remake the version.h header file if, and only if, the git ID has changed
 .PHONY: .FORCE
 .FORCE:
-generate_ver = ver="`{ $(srcdir)/scripts/version || echo '$(VERSION)' ; } | sed 's/.*/\x23define JQ_VERSION \"&\"/'`"
+generate_ver = ver="`{ $(srcdir)/scripts/version || echo '$(VERSION)' ; } | xargs printf '\043define JQ_VERSION \"%s\"\n'`"
 .remake-version-h: .FORCE
 	@ $(generate_ver); test "x`cat src/version.h 2>/dev/null`" = "x$$ver" || touch .remake-version-h
 src/version.h: .remake-version-h

